pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1564,7 +1564,7 @@ func (r *ReconcileClusterDeployment) mergePullSecrets(cd *hivev1.ClusterDeployme
 	globalPullSecretName := os.Getenv(constants.GlobalPullSecret)
 	var globalPullSecret string
 	if len(globalPullSecretName) != 0 {
-		globalPullSecret, err = controllerutils.LoadSecretData(r.Client, globalPullSecretName, constants.HiveNamespace, corev1.DockerConfigJsonKey)
+		globalPullSecret, err = controllerutils.LoadSecretData(r.Client, globalPullSecretName, controllerutils.GetHiveNamespace(), corev1.DockerConfigJsonKey)
```

Comment:
Shouldn't the global pull secret by in the namespace of the hive-operator? It is part of the operand.

Reply:
The two sentences here seem to conflict, hive-controllers is the operand right? As such I believe this is the right place for it.

This secret is created by an admin, added to HiveConfig (alongside targetNamespace), and we assume it's in targetNamespace. (right now) I believe it's only used by hive-controllers, and I don't think hive-controllers should need to be passed the operator's namespace, or read anything from it, it should be largely unaware of the operator's existence. Feels better here than the alternative.

Reply:
Sorry, I used the wrong terminology. I was thinking that the HiveConfig was the operand. My thought was that the admin would not be creating or touching the targetNamespace namespace at all. All of the admin configuration is done relative to HiveConfig and the hive-operator. The hive-operator is the one that creates the targetNamespace namespace. Is the expectation that the admin would be create that namespace instead? Or is the expectation that the admin would be create the global-pull-secret secret after the hive-operator create that namespace?

Reply:
The hive-operator is the one that can create the target namespace if it doesn't exist, but the admin can pre-create and add their global pull secret there prior to creating HiveConfig if desired, or update HiveConfig post-install.

Reply:
OK. I am fine with this.